src/main/java/com/sonyericsson/rebuild/RebuildAction.java

```diff
@@ -297,7 +297,13 @@ public void doConfigSubmit(StaplerRequest req, StaplerResponse rsp) throws Servl
                 }
             }
 
-            List<Action> actions = constructRebuildCause(build, new ParametersAction(values));
+            Set<ParameterValue> mergedValues = ImmutableSortedSet.copyOf(new Comparator<ParameterValue>() {
+                @Override
+                public int compare(ParameterValue o1, ParameterValue o2) {
+                    return o1.getName().compareTo(o2.getName());
+                }
+            }, Iterables.concat(values, paramAction.getParameters()));
```

Comment:
@amuniz Added the null check to be safe but struggling to see how 'paramAction' would be null at this stage based on - https://github.com/jenkinsci/rebuild-plugin/blob/master/src/main/java/com/sonyericsson/rebuild/RebuildAction.java#L206

The lack of null checks in the existing code seems to echo this - https://github.com/jenkinsci/rebuild-plugin/blob/master/src/main/java/com/sonyericsson/rebuild/RebuildAction.java#L417 

Either that or its just really fragile :)

cheers